code/modules/mod/modules/modules_supply.dm

```diff
@@ -0,0 +1,577 @@
+	allow_flags = MODULE_ALLOW_INCAPACITATED //Required so hands blocked doesnt block bombs
+	/// Time it takes us to complete the animation.
+	var/animate_time = 0.25 SECONDS
+	/// List of traits to add/remove from our subject as needed.
+
+/obj/item/mod/module/sphere_transform/on_activation()
+	if(!has_gravity(get_turf(src)))
+		to_chat(mod.wearer, "<span class='warning'>ERROR, NO GRAVITY!</span>")
+		return FALSE
+	. = ..()
+	if(!.)
+		return
+	playsound(src, 'sound/items/modsuit/ballin.ogg', 100, TRUE)
+	mod.wearer.add_filter("mod_ball", 1, alpha_mask_filter(icon = icon('icons/mob/clothing/modsuit/mod_modules.dmi', "ball_mask"), flags = MASK_INVERSE))
+	mod.wearer.add_filter("mod_blur", 2, angular_blur_filter(size = 15))
+	mod.wearer.add_filter("mod_outline", 3, outline_filter(color = "#000000AA"))
+	animate(mod.wearer, animate_time, pixel_y = mod.wearer.pixel_y - 4, flags = ANIMATION_PARALLEL)
+	mod.wearer.SpinAnimation(1.5)
+	ADD_TRAIT(mod.wearer, TRAIT_HANDS_BLOCKED, "metriod[UID()]")
+	ADD_TRAIT(mod.wearer, TRAIT_GOTTAGOFAST, "metroid[UID()]")
+	RegisterSignal(mod.wearer, COMSIG_MOB_STATCHANGE, PROC_REF(on_statchange))
+
+/obj/item/mod/module/sphere_transform/on_deactivation(display_message = TRUE, deleting = FALSE)
+	. = ..()
+	if(!.)
+		return
+	if(!deleting)
+		playsound(src, 'sound/items/modsuit/ballin.ogg', 100, TRUE, frequency = -1)
+	animate(mod.wearer, animate_time, pixel_y = mod.wearer.pixel_y + 4)
+	addtimer(CALLBACK(mod.wearer, TYPE_PROC_REF(/atom, remove_filter), list("mod_ball", "mod_blur", "mod_outline")), animate_time)
+	REMOVE_TRAIT(mod.wearer, TRAIT_HANDS_BLOCKED, "metriod[UID()]")
+	REMOVE_TRAIT(mod.wearer, TRAIT_GOTTAGOFAST, "metroid[UID()]")
+	UnregisterSignal(mod.wearer, COMSIG_MOB_STATCHANGE)
+
+/obj/item/mod/module/sphere_transform/on_use()
+	if(!lavaland_equipment_pressure_check(get_turf(src)))
+		to_chat(mod.wearer, "<span class='warning'>ERROR, OVER PRESSURE!</span>")
+		playsound(src, 'sound/weapons/gun_interactions/dry_fire.ogg', 25, TRUE)
+		return FALSE
+	return ..()
+
+/obj/item/mod/module/sphere_transform/on_select_use(atom/target)
+	. = ..()
+	if(!.)
+		return
+	var/obj/item/projectile/bomb = new /obj/item/projectile/bullet/reusable/mining_bomb(get_turf(mod.wearer))
+	bomb.original = target
+	bomb.firer = mod.wearer
+	bomb.preparePixelProjectile(target, get_turf(target), mod.wearer)
+	bomb.fire()
+	playsound(src, 'sound/weapons/grenadelaunch.ogg', 75, TRUE)
+	drain_power(use_power_cost)
+
+/obj/item/mod/module/sphere_transform/on_active_process()
+	animate(mod.wearer) //stop the animation
+	mod.wearer.SpinAnimation(1.5) //start it back again
+	if(!has_gravity(get_turf(src)))
+		on_deactivation() //deactivate in no grav
+
+/obj/item/mod/module/sphere_transform/proc/on_statchange(datum/source)
+	SIGNAL_HANDLER
+	if(!mod.wearer.stat)
+		return
+	on_deactivation()
+
+/obj/item/projectile/bullet/reusable/mining_bomb
+	name = "mining bomb"
+	desc = "A bomb. Why are you examining this?"
```

Comment:
```suggestion
	desc = "A bomb. Why are you staring at this?"
```
IC terms preferred for descriptions

Reply:
wait till you see the examine of pointing, but yup, will change it!